scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -861,7 +861,10 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       if (rangeFromStart.isEmpty) (startTokenPos, endTokenPos)
       else (skipTrivia(rangeFromStart), skipTrivia(rangeFromStart.reverse))
 
-    TokenStreamPosition(start, end + 1)
+    if (start == end && scannerTokens(start).is[Trivia])
+      TokenStreamPosition(start, end)
```

Comment:
what if start != end? does it ever make sense to start a tree with Whitespace?

Reply:
if start != end than trivia is already removed fully.

Reply:
got it... must be because it's not the start that has it but end.